src/catalog.js

```diff
@@ -181,15 +181,17 @@ angular.module('gettext').factory('gettextCatalog', function (gettextPlurals, ge
                     val = obj;
                 }
 
+                this.strings[language][key] = this.strings[language][key] || {};
                 // Expand single strings for each context.
                 for (var context in val) {
                     var str = val[context];
                     if (!angular.isArray(str)) {
-                        val[context] = [];
-                        val[context][defaultPlural] = str;
+                        this.strings[language][key][context] = [];
```

Comment:
The point is to keep the old values in `this.strings[language][key][context]`,
so either just to add to that. Or need an other loop to put all existing value in `this.strings[language][key][context]` to val. And then this.strings[language][key][context]=val;
The original problem is that:
`this.strings[language][key][context] = val` overwrite everything.